src/clj_jargon/item_info.clj

```diff
@@ -124,44 +131,33 @@
   (validate-path-lengths path)
   (str (long (.getTime (.getCreatedAt (.getObjStat cm-ao path))))))
 
-(defn- dir-stat
-  "Returns status information for a directory."
-  [cm ^String path]
-  (validate-path-lengths path)
-  (let [coll (collection cm path)]
-    {:id            path
-     :path          path
-     :type          :dir
-     :date-created  (long (.. coll getCreatedAt getTime))
-     :date-modified (long (.. coll getModifiedAt getTime))}))
-
-(defn- file-stat
-  "Returns status information for a file."
-  [cm ^String path]
+(defn file-size
+  "Returns the size of the file in bytes."
+  [{^IRODSFileSystemAO cm-ao :fileSystemAO} ^String path]
   (validate-path-lengths path)
-  (let [data-obj (data-object cm path)]
-    {:id            path
-     :path          path
-     :type          :file
-     :file-size     (.getDataSize data-obj)
-     :md5           (.getChecksum data-obj)
-     :date-created  (long (.. data-obj getCreatedAt getTime))
-     :date-modified (long (.. data-obj getUpdatedAt getTime))}))
+  (.getObjSize (.getObjStat cm-ao path)))
 
 (defn stat
   "Returns status information for a path."
-  [cm ^String path]
-  (validate-path-lengths path)
-  (cond
-   (is-dir? cm path)  (dir-stat cm path)
-   (is-file? cm path) (file-stat cm path)
-   :else              nil))
-
-(defn file-size
-  "Returns the size of the file in bytes."
   [{^IRODSFileSystemAO cm-ao :fileSystemAO} ^String path]
   (validate-path-lengths path)
-  (.getObjSize (.getObjStat cm-ao path)))
+  (let [objstat (.getObjStat cm-ao path)]
```

Comment:
Here's big improvement #1: we don't need to create full-on `DataObject` or `Collection` objects to get the information below, and at least the former is really slow. So now this just uses one much lighter call.